Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -513,6 +581,26 @@ void Interpreter::getStackTrace(JSCell* owner, Vector<StackFrame>& results, size
             return IterationStatus::Continue;
         }
 
+        if (Options::useAsyncStackTrace()) {
+            if (visitor->callee().isCell()) {
+                if (JSFunction* callee = jsDynamicCast<JSFunction*>(visitor->callee().asCell())) {
+                    JSGlobalObject* globalObject = callFrame->lexicalGlobalObject(vm);
+
+                    JSGenerator* generator = nullptr;
+                    if (callee == globalObject->promiseReactionJobFunction())
+                        generator = jsDynamicCast<JSGenerator*>(visitor->callFrame()->argument(3));
+                    else if (callee == globalObject->promiseReactionJobWithoutPromiseFunction())
+                        generator = jsDynamicCast<JSGenerator*>(visitor->callFrame()->argument(2));
```

Comment:
Yeah, this is not correct. StackVisitor can populate each frame's callee. And there is no guarantee whether each function gets inlined or not. If these functions are inlined, then CallFrame is the inliner's CallFrame, and arguments are their arguments.

Reply:
I changed it to prevent inlining using the `@neverInline` attribute added in https://github.com/WebKit/WebKit/pull/49873